packages/@aws-cdk/cdk/lib/cloudformation/output.ts

```diff
@@ -76,16 +81,16 @@ export class Output extends StackElement {
     this._value = props.value;
     this.condition = props.condition;
 
+    this.disableExport = props.disableExport !== undefined ? props.disableExport : false;
+
+    if (props.export && this.disableExport) {
+      throw new Error('Cannot set `disableExport` and specify an export name');
```

Comment:
This seems to contradict what the `props` documentation says. But TBH I prefer this... And I'd suggest replacing `disableExport` with something like `exportable` (semantic is flipped:  if `!exportable`, then you cannot provide - or have allocated - an `export` name).